monitor/filter_exception_decoder.py

```
@@ -358,160 +504,592 @@ def rx(self, text):
+
+    def _decode_batch(self, addrs, elf_path):
+        """Decode multiple addresses in a single addr2line call (-fiaC)."""
+        if not addrs:
+            return
+
+        addr_list = list(addrs)
+        enc = "mbcs" if IS_WINDOWS else "utf-8"
+        args = [self.addr2line_path, "-fiaC", "-e", elf_path] + addr_list
+
+        try:
+            raw = subprocess.check_output(args).decode(enc)
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)